lib/associations/associations.rb

```diff
@@ -8,7 +8,14 @@ def belongs_to(*args)
         options = our_args.extract_options!
         name = our_args.shift
         options = {:class_name => name.to_s.camelize }.merge(options)
-        klass = options[:class_name].constantize rescue nil
```

Comment:
`#safe_constantize` is still not applicable if the Autoloader is being used.

I added a test case that simulates the use of the autoloader.  The test fails on master, and passes on this branch.